ui/src/hooks/use-packages-by-product-suspense.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2024 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import { useSuspenseQueries } from '@tanstack/react-query';
+import { useMemo } from 'react';
+
+import {
+  UsePackagesServiceGetPackagesByRunIdKeyFn,
+  UseRepositoriesServiceGetOrtRunsByRepositoryIdKeyFn,
+} from '@/api/queries';
+import { useRepositoriesServiceGetRepositoriesByProductIdSuspense } from '@/api/queries/suspense';
+import { Package, PackagesService, RepositoriesService } from '@/api/requests';
+
+export function usePackagesByProductIdSuspense({
+  productId,
+}: {
+  productId: number;
+}): Package[] {
+  // Get all repositories of the product.
+  const { data: repositories } =
+    useRepositoriesServiceGetRepositoriesByProductIdSuspense({
+      productId,
+    });
+
+  // Get the newest ORT run for each repository for which the analyzer job has succeeded.
+  const runsForAnalyzerQueries = useSuspenseQueries({
+    queries: repositories.data.map((repository) => ({
+      queryKey: [
+        'runsForAnalyzerInProduct',
+        ...UseRepositoriesServiceGetOrtRunsByRepositoryIdKeyFn({
+          repositoryId: repository.id,
+          sort: '-index',
+        }),
+      ],
+      queryFn: async () => {
+        const runs = await RepositoriesService.getOrtRunsByRepositoryId({
+          repositoryId: repository.id,
+          sort: '-index',
+        });
+        for (let i = 0; i < runs.data.length; i++) {
+          const run = runs.data[i];
+          if (
+            run?.jobs.analyzer &&
+            (run.jobs.analyzer.status === 'FINISHED' ||
+              'FINISHED_WITH_ISSUES') &&
+            run.jobs.analyzer.finishedAt
+          ) {
+            return run;
+          }
+        }
+        return null;
+      },
+    })),
+  });
+
+  // Safety measure to ensure that only valid data is included in the array.
+  const ortRuns = useMemo(
+    () => runsForAnalyzerQueries.filter((query) => query.data).filter(Boolean),
+    [runsForAnalyzerQueries]
+  );
```

Comment:
I don't think this actually does anything, as for some reason the null check needs ot be done in the below useSuspenseQueries as well. So I believe this could be removed and have the below queries be a flatMap of runsForAnalyzerQueries.